plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/azure.ts

```diff
@@ -107,9 +107,7 @@ export function createPublishAzureAction(options: {
         );
       }
 
-      // blam: Repo contents is serialized into the path,
-      // so it's just the base path I think
-      const repoContentsUrl = remoteUrl;
+      const repoContentsUrl = `https://${host}/${organization}/${owner}/_git/${repo}`;
```

Comment:
I think I'm a bit unclear on what's the difference between this URL and the input `ctx.input.repoUrl`. With this change, the test makes less sense to me than it did before. 🤔  I think I'll let @benjdlambert have a look at this one. I think the test can be clarified with better naming of things, and same thing with the parameter descriptions above in this file

Reply:
The `remoteUrl` that I'm getting back from azure has this format: `https://org.visualstudio.com/owner/_git/repo`. This of course won't work with registering of the component as the token defined in the integrations has the `dev.azure.com` domain and not `visualstudio.com`

Reply:
I'm not sure you you need this change.

I think by adding.

```yaml
integrations:
    azure:
        host: org.visualsstudio.com
        token: ...
```

Should work right? It's a little strange that Azure returns a different host, but we don't want to mutate what the provider returns as it could quite rightly return a different path from what you send in.

Reply:
I'm just a bit concerned about storing the components using the "old" URL. According to this post, they'll eventually migrate which would mean that we'd have to change all the URLs after that.
https://developercommunity.visualstudio.com/t/switch-existing-organizations-to-use-the-new-domai/765309

Reply:
I'm not really sure that this is the place to do it for everyone though. This could accidentally break other users that haven't planned on migrating yet and are still using the old URL. It would be the opposite problem for them with this fix, they they wouldn't be able to import things to the catalog when their credentials are stored under `org.visualstudio.com`?

Looking in the URL's in the post, there's this which might solve your issue?
https://docs.microsoft.com/en-us/azure/devops/release-notes/2018/sep-10-azure-devops-launch#switch-existing-organizations-to-use-the-new-domain-name-url

If you've already done that, I'd recommend reaching out the DevOps support to ask them why their API is returning the old URL when you've already told them not to?

Reply:
Yes we've done that. I'll reach out to Azure to debug it further. Do you want to wait for feedback from Azure team or abandon this PR already?

Reply:
It's entirely up to you - i'm happy to keep this around and come up with a nicer solution if they come back with a new config param or something to pass through the client, but maybe we can raise that seperately?

Reply:
Let's keep it open and I'll get back here hopefully soon.

Reply:
> @nadworny @benjdlambert We are encountering the same issue as well.  It didn't work by adding
    azure:
        host: org.visualsstudio.com
        token: ......
Is there any solution or workaround for solving this problem? Thx

Reply:
@ttpp123 hey, we've changed the default domain name to the new one dev.azure.com. You'd need to ask your admin to do that.

Reply:
@ttpp123 I ran into the same issue and had no chance to change the to go with new URL:
Following approach worked:
```yaml
   - id: register
      name: Register
      action: catalog:register
      input:
        repoContentsUrl: ${{"https://dev.azure.com/danfoss/" + (parameters.repoUrl | parseRepoUrl)["owner"] + "/_git/" + (parameters.repoUrl | parseRepoUrl)["repo"] }}
        catalogInfoPath: "/catalog-info.yaml"
```